src/adapter/src/coord.rs

```diff
@@ -154,11 +154,16 @@ pub enum Message<T = mz_repr::Timestamp> {
     SinkConnectionReady(SinkConnectionReady),
     SendDiffs(SendDiffs),
     WriteLockGrant(tokio::sync::OwnedMutexGuard<()>),
-    AdvanceTimelines,
-    AdvanceLocalInput(AdvanceLocalInput<T>),
-    GroupCommit,
+    GroupCommitInitiate,
+    GroupCommitApply(
+        T,
+        Vec<CompletedClientTransmitter<ExecuteResponse>>,
+        Option<OwnedMutexGuard<()>>,
+    ),
```

Comment:
Nit, but it would be helpful to have comments on these variants. I'm sort of guessing what `Apply` means, for example, maybe "it finished"? But, this enum in particular calls out for doccomments explaining what each message means.

Reply:
Good idea, btw I was having a hard time coming up with a good name for this variant and settled on "Apply" if you have a better idea then let me know.

Reply:
The idea is that "Apply" makes a group commit visible by doing the following:
1. Advancing the timeline.
2. Sending responses to clients.
3. Updating read holds (This is purely for convenience and we could do this elsewhere).